server/lib/flow-metrics.js

```diff
@@ -4,29 +4,56 @@
 
 var crypto = require('crypto');
 
-/**
- *
- * @param key String flow_id_key
- * @param userAgent String request user agent string
- * @returns {{flowBeginTime: number, flowId: string}}
- */
-module.exports = function flowEventData(key, userAgent) {
-  var flowId = crypto.randomBytes(16).toString('hex');
-  var flowBeginTime = Date.now();
+const SUBSTRING_LENGTH = 32;
```

Comment:
naming nit: perhaps we can extract `SALT_LENGTH = 16` into a constant as well, and call this something like `SALT_LENGTH_HEX = SALT_LENGTH * 2`.